### DISCUSSION TITLE
Tray icon missing on Ubuntu Budgie 19.04

### DISCUSSION THREAD

**Discussion by drohm** [borgbase/vorta#851] [General]
When installing the flatpak version, after starting Vorta, it shows up in the tray but the icon is missing:

![vorta-missing-tray-icon](https://user-images.githubusercontent.com/184959/60107281-a262fa80-9734-11e9-85b0-12a790dd740e.png)

**To Reproduce**
1. Install flatpak and launch vorta.

**Desktop (please complete the following information):**
 - OS: Ubuntu Budgie 19.04

**Comment by Hofer-Julian** [comment]
Could please install Vorta via `pip3 install --user vorta` (you might have to execute `sudo apt install python3-pip` first) and find out if the icon is there missing too?

**Comment by drohm** [comment]
@Hofer-Julian On my previous operating install (I had to reinstall yesterday) I used the pip3 installation and the tray icon showed up perfectly.  On this OS install I tried the flatpak and the icon is missing.

**Comment by Hofer-Julian** [comment]
Okay!

Please run `flatpak run com.borgbase.Vorta --foreground` and post the output of the terminal here.

**Comment by drohm** [comment]
Here's the output:
```
Gtk-Message: 11:47:48.582: Failed to load module "canberra-gtk-module"
Gtk-Message: 11:47:48.583: Failed to load module "canberra-gtk-module"
2019-06-25 11:47:48,603 - vorta.i18n - DEBUG - Loading translation failed for ['en', 'en-US', 'en-Latn-US'].
2019-06-25 11:47:48,606 - apscheduler.scheduler - INFO - Scheduler started
2019-06-25 11:47:48,608 - apscheduler.scheduler - INFO - Added job "VortaScheduler.create_backup" to job store "default"
2019-06-25 11:47:48,608 - vorta.scheduler - DEBUG - New job for profile EzyRhinox was added.
libGL error: No matching fbConfigs or visuals found
libGL error: failed to load driver: swrast
2019-06-25 11:47:49,236 - vorta.borg.borg_thread - INFO - Running command /app/bin/borg --version
```

**Comment by Hofer-Julian** [comment]
I didn't see the error message I expected, but please still try to run:
`flatpak run --own-name='org.kde.*' com.borgbase.Vorta`

**Comment by drohm** [comment]
The only output from that was:
```
Gtk-Message: 12:42:12.224: Failed to load module "canberra-gtk-module"
Gtk-Message: 12:42:12.224: Failed to load module "canberra-gtk-module"
```

**Comment by Hofer-Julian** [comment]
But no tray icon?

**Comment by drohm** [comment]
Correct, still no icon.

**Comment by Hofer-Julian** [comment]
Okay, I will test it myself on Ubuntu Budgie 19.04 when I have time

**Comment by Hofer-Julian** [comment]
I could reproduce the issue on Ubuntu Budgie 19.04.

**Comment by Vinno97** [comment]
I am also having issues with the tray icon while using Flatpak, though I am using Manjaro Gnome instead of Ubuntu Budgie. 
When installed using `pip install vorta`, the tray icon works as expected, but not when it's installed via flatpak.

I do have some differences to @drohm's problems, though. 
1. Vorta reports the error `failed to register service "org.kde.StatusNotifierItem-3-1"` when ran via flatpak with the `--foreground` flag. 
1. Using `flatpak run --own-name='org.kde.*' com.borgbase.Vorta` *did* fix my issues. 

---

There is another difference between the pip and the flatpak version that I found, namely that the flatpak version also prints the error `Qt: Session management error: None of the authentication protocols specified are supported`. 

Perhaps these findings help you in finding the root cause.

**Comment by drohm** [comment]
Quick note for this issue, I upgraded to Ubuntu Budgie 19.10 and the issue still exists.

**Comment by elhutchi** [comment]
In v0.6.24 (via Flatpak test channel on Debian), I still had to run with `--own-name='org.kde.*'` for the tray icon to show up. Should [this](https://github.com/borgbase/vorta/blob/facfbf29f4e2f00c970420d10449f1e7ec240c9d/flatpak/com.borgbase.Vorta.json#L22) be `org.kde.*` instead of `org.KDE.*`?

**Comment by Hofer-Julian** [comment]
> In v0.6.24 (via Flatpak test channel on Debian), I still had to run with `--own-name='org.kde.*'` for the tray icon to show up. Should [this](https://github.com/borgbase/vorta/blob/facfbf29f4e2f00c970420d10449f1e7ec240c9d/flatpak/com.borgbase.Vorta.json#L22) be `org.kde.*` instead of `org.KDE.*`?

@elhutchi yes, I think this change is necessary as well. The next flatpak update will include this change. After you've updated, please report if it solves your problem.

**Comment by elhutchi** [comment]
@Hofer-Julian Thanks, I can confirm https://github.com/flathub/com.borgbase.Vorta/pull/17 has fixed the tray icon in GNOME, but I realized this issue was actually for Ubuntu Budgie. Sorry for hijacking it.

**Comment by drohm** [comment]
@Hofer-Julian I tried the solution that @elhutchi mentioned but I still cannot get the icon to show.  Have you been able to get the icon to show properly in the tray using this method?

**Comment by Hofer-Julian** [comment]
@drohm do you use the newest flatpak version?
Please run` flatpak update `in the commandline

**Comment by drohm** [comment]
@Hofer-Julian I'm using the latest version, 0.6.23.

**Comment by Hofer-Julian** [comment]
If you haven't yet, still update. The version number hasn't changed

**Comment by elhutchi** [comment]
@Hofer-Julian I had to install the latest version of 0.6.23 by running the command in the pull request above. Otherwise the latest version from the stable branch didn't work without the extra `--own-name='org.kde.*'` argument.

And when I run `flatpak remote-info flathub com.borgbase.Vorta` the last update was related to the previous pull request. Is flathub updated? (I'm not sure how the publishing works.)

**Comment by drohm** [comment]
@Hofer-Julian I have been running `flatpak update` daily and it's not showing any updates (Nothing to do.)

**Comment by drohm** [comment]
@elhutchi, could you provide the commands to run to get the version you're running so I can test that as well?  I uninstalled the version from the main flathub repo so I can try this, but wasn't sure which commands to run.

**Comment by elhutchi** [comment]
I think all I had to do was run the command in [this comment](https://github.com/flathub/com.borgbase.Vorta/pull/17#issuecomment-602101061):

```
flatpak install --user https://dl.flathub.org/build-repo/16469/com.borgbase.Vorta.flatpakref
```

**Comment by drohm** [comment]
@elhutchi Yea, I tried that but its not working for me:

```
error: Can't load uri https://dl.flathub.org/build-repo/16469/com.borgbase.Vorta.flatpakref: Server returned status 404: Not Found
```

**Comment by matty-fb** [comment]
I'm having the same notification icon issue in Xubuntu 20.04. I'm running Vorta 0.8.3 via flatpak, and launching with `--own-name='org.kde.*'` does not fix the icon. The notification icon _does_ work properly when installed via pip3.
